src/transforms/groupby.js

```diff
@@ -35,6 +35,35 @@ exports.attributes = {
             'with `x` [1, 3] and one trace with `x` [2, 4].'
         ].join(' ')
     },
+    nameformat: {
+        valType: 'string',
+        description: [
+            'Pattern by which grouped traces are named. If only one trace is present,',
+            'defaults to the group name (`"%{group}"`), otherwise defaults to the group name',
+            'with trace name (`"%{group} (%{trace})"`). Available escape sequences are `%{group}`, which',
+            'inserts the group name, and `%{trace}`, which inserts the trace name. If grouping',
+            'GDP data by country when more than one trace is present, for example, the',
+            'default "%{group} (%{trace})" would return "Monaco (GDP per capita)".'
+        ].join(' ')
+    },
+    groupnames: {
```

Comment:
> setting `groupnames[i].name` is equivalent to changing the corresponding `groups` item.

Not quite - the name will be set by `nameformat` - ie a combination of the group value and the trace name by default - but when you want to override that (perhaps by typing something specific in the editable legend) you'll need this.

> We could instead declare `groupnames` as a `data_array` and expecting its length to be equal to `groups.length`, but that would lead to some (maybe a lot of) duplication.

Or ambiguity, if the names were not perfectly duplicated. What would that mean?

> Maybe we could `groupname` attribute to the `styles` container below instead which already expects a target group?

That sounds great! so each entry in `styles` would have `target` and optionally `value` and/or `name`? No need to have two separate containers both keyed on the group value.